OpenSim/Common/Component.h

```diff
@@ -3050,11 +3052,12 @@ void Socket<C>::finalizeConnection(const Component& root) {
             "root component. Did you intend to add '" +
             rootOfConnectee.getName() + "' to '" + myRoot.getName() + "'?");
 
-        std::string objPathName = connectee->getAbsolutePathString();
-        std::string ownerPathName = getOwner().getAbsolutePathString();
-        // otherwise store the relative path name to the object
-        std::string relPathName = connectee->getRelativePathName(getOwner());
-        updConnecteePathProp().setValue(0, relPathName);
+        std::string connecteePath = connectee->getRelativePathName(getOwner());
+        // If the relative path start with ".." the use an absolute path
+        // instead.
+        if (connecteePath.size() >= 2 && connecteePath.substr(0, 2) == "..")
```

Comment:
I doubt this would really be an issue, but there shouldn't be a way to form a "correct" path such that it's something like
`..someCharsBeforeTheFirstSlash/foo/bar`, right?

Reply:
Good question. Based on these lines:

https://github.com/opensim-org/opensim-core/blob/083fcc4efcbf47839bdc1906fcfa634df14fe148/OpenSim/Common/Component.cpp#L178-L180

...I think it would be possible to have such a path and it might mostly function without errors. Perhaps we should add to the path validation that if a path element is not "." or ".." then it cannot contain dots.

In this case, an "absolute path" will be used (e.g., `/bar/foo/..someChars/biz/faz`) and the connectee should still be found.

I suppose the issue is not so different from that for filesystems: it's valid to have `.` in a filename.

Reply:
Maybe the way to be sure then is to use the `Path` class to check if the first element of the `Path` is exactly `..`? I guess it just depends on how much we want this to be like a filesystem and/or accept `.` in a path. Separately, I don't think this is critical for this PR.

Reply:
I've updated the PR to use the `Path` class. Let me know what you think.

Reply:
LGTM

Reply:
Thanks.